src/h2load.cc

```diff
@@ -730,20 +758,37 @@ void Worker::run() {
   ev_run(loop, 0);
 }
 
+//TODO make this work with other stats
 namespace {
 double within_sd(const std::vector<std::unique_ptr<Worker>> &workers,
                  const std::chrono::microseconds &mean,
-                 const std::chrono::microseconds &sd, size_t n) {
+                 const std::chrono::microseconds &sd, size_t n, TimeStatType type) {
   auto upper = mean.count() + sd.count();
   auto lower = mean.count() - sd.count();
   size_t m = 0;
   for (const auto &w : workers) {
+    const auto &stat = w->stats;
     for (const auto &req_stat : w->stats.req_stats) {
       if (!req_stat.completed) {
         continue;
       }
-      auto t = std::chrono::duration_cast<std::chrono::microseconds>(
-          req_stat.stream_close_time - req_stat.request_time);
+      std::chrono::steady_clock::time_point stat1;
+      std::chrono::steady_clock::time_point stat2;
+      switch(type){
+        case STAT_REQUEST: 
+          stat1 = req_stat.stream_close_time;
+          stat2 = req_stat.request_time;
+          break;
+        case STAT_CONNECT:
+          stat1 = stat.connect_time;
+          stat2 = stat.start_time;
+          break;
+        case STAT_FIRST_BYTE:
+          stat1 = stat.time_to_first_byte;
+          stat2 = stat.start_time;
+          break;
+      }
+      auto t = std::chrono::duration_cast<std::chrono::microseconds>(stat1 - stat2);
```

Comment:
ah yeah. i think it might make sense for me to make a separate within_sd for STAT_CONNECT and STAT_FIRST_BYTE cases.